src/index.ts

```diff
@@ -193,24 +195,34 @@ function startApp(appSchema, path: string) {
           userAgent,
         }
 
+        const validationRules = [principalFieldDirectiveValidation]
+        if (QUERY_DEPTH_LIMIT)
+          validationRules.push(depthLimit(QUERY_DEPTH_LIMIT))
+
         return {
           schema: appSchema,
           graphiql: true,
           context,
           rootValue: {},
-          // FIXME: This needs to be updated as per the release notes of graphql-js v14
           formatError: graphqlErrorHandler(enableSentry, {
             req,
             // Why the checking on params? Do we reach this code if params is falsy?
             variables: params && params.variables,
             query: (params && params.query)!,
           }),
-          validationRules: QUERY_DEPTH_LIMIT
-            ? [depthLimit(QUERY_DEPTH_LIMIT)]
-            : null,
-          extensions: enableRequestLogging
-            ? () => fetchLoggerRequestDone(requestID)
-            : undefined,
+          validationRules,
+          extensions: ({ document, result }) => {
+            const principalFieldExtensions = principalFieldDirectiveExtension(
+              document,
+              result
+            )
+
+            const requestLoggerExtensions = enableRequestLogging
+              ? fetchLoggerRequestDone(requestID)
+              : {}
+
+            return { ...principalFieldExtensions, requestLoggerExtensions }
```

Comment:
Pairing with @ds300 on the graphql issue we're seeing now in Emission -- Looks like this needs to be `return { ...principalFieldExtensions, ...requestLoggerExtensions }` as `res.json.extensions` is missing the `requests` object.